### ISSUE TITLE
Formula range queries

### ISSUE THREAD

**Issue by meowcat** [mwang87/MassQueryLanguage_Documentation#6]
Hi,

I'm looking forward to applying MassQL in compound ID workflows. You might know that a vaguely similar approach, MFQL, exists for lipidomics, but to my knowledge there is no implementation that works independently from a complete workflow. I was hoping to "translate" some existing MFQL queries to MassQL. However, one feature of MFQL is currently not yet available as far as I can see: specifying a range of formulas rather than a single formula. e.g.

`DEFINE PR = 'C[17..25] H[30..70] O[7] P[1]' WITH DBR = (1.5,7.5), CHG = -1;`

https://wiki.mpi-cbg.de/lipidxscr/images/6/6f/Neg_bovine_heart_LPA.mfql
https://wiki.mpi-cbg.de/lipidx/MFQL_library

Do you have any intentions of going in this direction in the future?

**Comment by mwang87** [comment]
Hey Michael,

Thanks for the question and definitely, it seemed that MFQL is a great resource for the lipidomics kinds of queries you point out. At least when I was creating MassQL lipidomics were not specifically on my mind for these kinds of formula space queries. I've been thinking a bit about it, but I don't have a good sense for the right kind of abstraction that would be appropriate for MassQL since I had intended this to be a more general tool. However, I invite you to be part of the team to help us think about it. For the time being, I think it should be relatively easy to translate from MFQL to MassQL by explicitly instantiating the queries, in this case for all the formulas here, it would be 8 (for the carbons) * 40 (for the hydrogens) which would make 320 total explicit queries. Not the most efficient way, but it does work and should be relatively fast as well. 

Swing by office hours and would love to work together to make things better. As always, its a work in progress but at the very least its helped quite a few of my colleagues. 

Best,

Ming

**Comment by meowcat** [comment]
Hi Ming,
> For the time being, I think it should be relatively easy to translate from MFQL to MassQL by explicitly instantiating the queries,

Maybe generating the queries is not a bad idea overall, since that would resolve the other issue - naming the results, which is a concept that MassQL isn't really intended to cover. I will look and see if I find a good way of formulating it.

**Comment by mwang87** [comment]
Definitely! Give it a go, and we have a mechanism to specify multiple queries using the ||| separator. One thing we also list in the output is the query index number so you know which result goes with which query. Though one issue is scaling to thousands of queries at once because of the limits on the commandline length, so if you want to down this route, we can make commandline tools more robust.
